protomap/proto.go

```diff
@@ -536,6 +536,10 @@ func ProtoFromPaths(p proto.Message, vals map[*gpb.Path]interface{}, opt ...Unma
 		return fmt.Errorf("invalid protobuf message prefix supplied in options, %v", err)
 	}
 
+	return protoFromPathsInternal(p, vals, valPrefix, protoPrefix, hasIgnoreExtraPaths(opt))
+}
+
+func protoFromPathsInternal(p proto.Message, vals map[*gpb.Path]any, valPrefix, protoPrefix *gpb.Path, ignoreExtras bool) error {
 	schemaPath := func(p *gpb.Path) *gpb.Path {
```

Comment:
Factor this out as its own helper function since it's not dependent on any function-scope variables?

Reply:
I see it's there in the next PR

Reply:
ACK, yes, apologies. I was thinking to do the same thing once I realised it became fully parameterised.